backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/MeetingService.cs

```diff
@@ -24,26 +23,73 @@ public MeetingService(EasyMeetsCoreContext context, IMapper mapper, IUserService
             _emailSender = emailSender;
         }
 
-        public async Task<List<MeetingThreeMembersDTO>> GetThreeMeetingMembersAsync(long? teamId)
+        public async Task<List<MeetingSlotDTO>> GetMeetingMembersByNumberOfMembersToDisplayAsync(long? teamId, int numberOfMembers)
         {
             if (teamId is null)
             {
-                return new List<MeetingThreeMembersDTO>();
+                return new List<MeetingSlotDTO>();
             }
 
             var team = await _context.Teams.FirstOrDefaultAsync(team => team.Id == teamId) ?? throw new KeyNotFoundException("Team doesn't exist");
 
             var meetings = await _context.Meetings
+                .Where(meeting => meeting.TeamId == team.Id)
                 .Include(m => m.AvailabilitySlot)
-                    .Include(s => s!.ExternalAttendees)
+                .Include(s => s!.ExternalAttendees)
                 .Include(meeting => meeting.MeetingMembers)
                     .ThenInclude(meetingMember => meetingMember.TeamMember)
                     .ThenInclude(teamMember => teamMember.User)
-                .Where(meeting => meeting.TeamId == team.Id)
-                .ToListAsync();
+                .Select(x =>
+                    new MeetingSlotDTO
+                    {
+                        Id = x.Id,
+                        Location = x.LocationType.ToString(),
+                        MeetingCount = x.MeetingMembers.Count,
+                        MembersTitle = CreateMemberTitle(x),
+                        MeetingTitle = x.Name,
+                        MeetingDuration = $"{x.Duration} min",
+                        MeetingTime = $"{x.StartTime.Hour}:{x.StartTime.Minute:00} - " +
+                            $"{x.StartTime.AddMinutes(x.Duration).Hour}:{x.StartTime.AddMinutes(x.Duration).Minute:00}",
+                        MeetingLink = x.MeetingLink,
+                        MeetingMembers = GetAllParticipants(x, numberOfMembers)
+                    })
+            .ToListAsync(); 
+
+            return meetings;
+        }
 
-            var mapped = _mapper.Map<List<MeetingThreeMembersDTO>>(meetings);
-            return mapped;
+        private static List<UserMeetingDTO> GetAllParticipants(Meeting meeting, int numberOfMembers)
+        {
+            var slotMembers = meeting.MeetingMembers
+                .Take(numberOfMembers)
```

Comment:
You need to take participants after union

Reply:
Agree with you. Fixed it.